### PR TITLE
Streaks and Dynamic XP

### PR DESCRIPTION
**Pull request by KashGiannis34**
## Solo Leveling the Leveling System

Issue Number(s): #157

What does this PR change and why?

Added backend functionality for streaks and dynamic xp reqs for each level. Not much was implemented on the frontend since this was a BE ticket but i did add info about the current streak and stuff like that on the store page.

### Checklist

- [ ] Requirements have been implemented
- [ ] Acceptance criteria is met
- [ ] Database schema docs have been updated or are not necessary
- [ ] Code follows design and style guidelines
- [ ] Commits follow guidelines (concise, squashed, etc)
- [ ] Relevant reviewers (Senior Dev/EM/Designers) have been assigned to this PR

### Critical Changes

- Database change/migration to run
- Environment config change
- Breaking API change

### Related PRs

- #number_of_pr

### Testing

Enumerate steps to test the functionality of your ticket. This should include edge cases and testing of error handling, if applicable.

### CODE REVIEW

**Comment by rgodha24** [comment]
i think we are gonna have a bit more of a longer conversation abt what to do with streaks and maybe push implementation to next semester? 
if u have time, can you make a pr with like just the 4 line change to tutorial logic, and we can leave this as a draft for next sem to deal with please?

File: src/types/inventory.ts [hunk 1]
```diff
@@ -8,6 +8,12 @@ export enum ItemType {
   OCCUPATION = "occupation",
 }
 
+export enum ItemPrice {
+  CHEAP = 10,
+  MEDIUM = 25,
+  EXPENSIVE = 50,
```
Comment (rgodha24) [inline src/types/inventory.ts]: need to talk abt this too
also, typescript enums are [kinda an antipattern](https://www.totaltypescript.com/why-i-dont-like-typescript-enums)

File: src/utils/constants.ts [hunk 2]
```diff
@@ -1,3 +1,11 @@
-export const LEVEL_THRESHOLD = 100;
+export const calculateXPForLevel = (level: number): number => {
+  return 90 + level * 10;
+};
+
```
Comment (rgodha24) [inline src/utils/constants.ts]: i feel like this doesnt rly need to be a helper

File: src/services/medication.ts [hunk 3]
```diff
@@ -201,13 +202,28 @@ export default class MedicationService {
       throw new NotFoundError("No pet found for the user");
     }
 
-    // add food to pet
+    const petWithId = existingPet as WithId<Pet>;
+    const streakUpdate = calculateStreakUpdate(petWithId, now);
+
     await PetDAO.updatePetByUserId(userId, {
       food: existingPet.food + FOOD_INC,
+      coins: existingPet.coins + streakUpdate.coinsAwarded + 10,
```
[the remark this answers is not in the document]
Comment (KashGiannis34) [reply src/services/medication.ts]: the tutorial gave +10 coins after the user logged the practice medication so i applied the same logic to the regular functionality
